src/details/ArborX_DetailsBatchedQueries.hpp

```diff
@@ -47,32 +47,34 @@ struct BatchedQueries
   // indirection when recording results rather than using that function at
   // the end.  We decided to keep reversePermutation around for now.
 
-  template <typename ExecutionSpace, typename Predicates>
+  template <typename ExecutionSpace, typename Predicates,
+            typename SpaceFillingCurve>
   static Kokkos::View<unsigned int *, DeviceType>
-  sortQueriesAlongZOrderCurve(ExecutionSpace const &space,
-                              Box const &scene_bounding_box,
-                              Predicates const &predicates)
+  sortPredicateAlongSpaceFillingCurve(ExecutionSpace const &space,
+                                      SpaceFillingCurve const &curve,
+                                      Box const &scene_bounding_box,
+                                      Predicates const &predicates)
   {
     using Access = AccessTraits<Predicates, PredicatesTag>;
     auto const n_queries = Access::size(predicates);
 
-    Kokkos::View<unsigned int *, DeviceType> morton_codes(
+    using OrderValueType =
```

Comment:
This name does not make sense to me. Could you please rename it to something like `CurveValueType` or `SFCValueType`? In `ArborX_DetailsTreeConstruction.hpp` you used `LinearOrderingValueType`, which I think is also fine.